### DISCUSSION TITLE
How to configure default team which would be added each time a new repository is created?

### DISCUSSION THREAD

**Discussion by kmadof** [community/community#62373] [Repositories] [answered]
### Select Topic Area

Question

### Body

I want to configure one or more default teams which would be added each time a new repository is created. Is this possible?

**Comment by skiks** [comment, marked as answer]
GitHub does not provide a built-in feature to automatically assign a default team to every new repository created. However, you can achieve this using GitHub's API and a simple script that runs whenever a new repository is created.

Here's a basic outline of how you could set this up:

Create a Webhook: In your organization settings, you can create a new Webhook. This Webhook should be configured to listen for the repository event. This event is triggered whenever a repository is created, deleted, made public, or made private.

Create a Server to Listen for the Webhook: You'll need a server that can listen for the incoming Webhook requests. This server could be a small application written in a language like Python, Node.js, or Ruby. When the server receives a request, it should parse the event and extract the necessary information, like the new repository's name and the organization it belongs to.

Use the GitHub API to Assign the Team: Once the server has the necessary information, it can make a request to the GitHub API to assign a team to the repository. The endpoint for this would be PUT /orgs/{org}/teams/{team_slug}/repos/{owner}/{repo}. You'll need to replace {org}, {team_slug}, {owner}, and {repo} with your organization's name, the team's slug, the repository owner's username, and the new repository's name, respectively.
